### DISCUSSION TITLE
How do you typically structure your project if it includes both frontend and fastapi?

### DISCUSSION THREAD

**Discussion by BramVanroy** [fastapi/fastapi#4344] [Questions] [answered]
I've been wondering about what the typical set-up is if you have both the front-end and the back-end API in the same repository that make use of completely different frameworks. For example, I use fastAPI as my back-end but react as my front-end. So how do I structure my project? What is the most common way of doing this? [In the documentation](https://fastapi.tiangolo.com/tutorial/bigger-applications/), I found how to structure fastAPI projects, but I wonder how to include front-end directory organization in that picture.

Currently, in [one of my projects](https://github.com/BramVanroy/ASTrED-Demo) I just put them in their separate folders (frontend/backend) and that's that. But maybe you would normally do this differently, i.e. with completely different repositories for the front and back-end? Or no subdirectories at all and everything in the root directory?

Happy to hear how you all do this!

**Comment by fabioueno** [comment, marked as answer]
Hey, @BramVanroy! I'm sorry for the late reply!

I hope this information can still be useful for you more than a year later. 😬

You mentioned how to structure FastAPI projects in the documentation. I'm not sure how old/new it is, but there is a section (and a template!) that shows what an application with FastAPI in the backend with Vue in the frontend would look like! You can [check it out here](https://fastapi.tiangolo.com/project-generation)! The repository with the code is this one: https://github.com/tiangolo/full-stack-fastapi-postgresql.

Some people mentioned it's a little outdated and suggested some other templates, that may interest you:
- https://github.com/isakbosman/full-stack-fastapi-react-postgres-boilerplate
- https://github.com/whythawk/full-stack-fastapi-postgresql

The second one seems to be the most promising, but I've never used any of them, to be honest.

Hope that helps! 😉

**Comment by caeser1996** [comment]
The way you organize your project depends on the size and complexity of your project, as well as your personal preference and team's conventions.

There are a few common ways to organize a project with both front-end and back-end components:

Separate folders within the same repository: You can have separate folders for the front-end and back-end within the same repository. This makes it easy to manage and track changes to both parts of the project.

Separate repositories: If the front-end and back-end are large and complex, you may want to keep them in separate repositories. This allows for better isolation and can make it easier to manage and track changes to each part of the project.

Monorepo: You can also keep both the front-end and back-end components within a single repository. This is called a monorepo. A monorepo allows you to manage all parts of the project from a single location, but can also lead to complex version control.

Ultimately, the best way to organize your project is one that works well for you and your team. You can start with one method and change it as needed as your project grows.
